src/main/java/org/commcare/formplayer/application/FormSubmissionController.java

```diff
@@ -252,6 +263,44 @@ private SubmitResponseBean processFormXml(FormSubmissionContext context) throws
         return context.success();
     }
 
+    private MultiValueMap<String, Object> getMultiPartFormBody(FormSession formSession) throws IOException {
+        MultiValueMap<String, Object> body = new LinkedMultiValueMap<>();
+
+        // Add any media files associated with the form
+        Path mediaDirPath = formSession.getMediaDirectoryPath(restoreFactory.getDomain(),
+                restoreFactory.getUsername(), restoreFactory.getAsUsername(), storageFactory.getAppId());
+        File mediaFile = mediaDirPath.toFile();
+        if(mediaFile.exists()) {
+            for (File file : Objects.requireNonNull(mediaDirPath.toFile().listFiles())) {
+                String contentType = FileUtils.getContentType(file);
+                if (!StringUtils.hasText(contentType)) {
+                    contentType = "application/octet-stream";
+                }
+                HttpEntity<Object> filePart = createFilePart(file.getName(), file.getName(),
+                        file, contentType);
+                body.add(file.getName(), filePart);
```

Comment:
I think there should be some additional validation before we submit these files in case there are other files in this directory for some reason. I'm thinking:
* max number of attachments
* max attachment size

Reply:
Sounds good, Would `10` be a good number for max no. of attachments ? Or we want to set it to much higher like `100` which more strongly signifies that something has gone wrong with the media directory.

Reply:
We should make it configurable via the properties file with a sane default. 10 seems like it could be too low - maybe 50? Or how about limiting on both number and total size - 50 attachments or total size < 50MB?

@orangejenny any thoughts about the max number of attachments to allow per form submission?

Reply:
50 sounds like a good threshold to go with. We do a size check per file while saving media, should we go with the same size check per file while submission as well so that it's more consistent ?

Reply:
Added a threshold of 50 and replicated normal file validations as well to protect against any malicious files - https://github.com/dimagi/formplayer/pull/1209/commits/e6bca3951eb723ab90c69072db79961dd2f22583